proposals/0007-rn-versions-lifecycle-support.md

```diff
@@ -0,0 +1,69 @@
+# RFC: React Native Versions Lifecycle & Support
+
+
+
+>As we are finding ways to better support React Native community, we are moving towards supporting a limited amount of versions of React Native better.
+
+
+The React Native Core team is working to strike a balance between the speed of innovation with the amount of regular change to the platform. We understand that each release brings new value and has an adoption cost to the community.
+This document describes how we intend to formally balance these concerns and provide clear and consistent support over time.
+React Native team will work on best-effort basis to release a new version of React Native every four months. And will focus on speeding up the release cycle and put considerable effort into ensuring a seamless upgrade path.
+This document describes the level of support we can provide and which set of versions is supported.
+
+## Glossary
+
+Below we are defining terms used in this document:
+
+* **stable** **version** - Any version that doesn’t have a -alpha, -beta, -RC postfix in the version name.
```

Comment:
here we're defining stable as "what it isn't", based on a property of the release tag, rather than what "stable" means. What guarantees exist in stable (i.e. what is the stability guarantee being made?) that isn't being made for non-stable releases?

Reply:
Thanks for your suggestion.

> here we're defining stable as "what it isn't", based on a property of the release tag, rather than what "stable" means.

Did you have a specific wording in mind for this?

Reply:
What I'd like to see happen here is some guarantee along the lines of e.g.: "there are no ABI breaks or other breaking changes within a stable release".

Reply:
> what is the stability guarantee being made?

tl;dr: There are no stability guarantees between versions (this was never the case). Breaking changes are mentioned inside the release notes of each releases in the "Breaking" section. We're not moving to full semver in this policy.

To clarify: the scope of this policy is just to write down what we've been informally doing in the recent past. Should we move versioning or provide further guarantess between versions, we're going to update the policy to mention it.

Reply:
understood - can you add a sentence to that effect, since this document talks about what the different release tags mean? e.g. "Note that despite the `-stable` tag, there are no stability guarantees. Breaking changes will be announced in each release's notes."

Reply:
I would refrain from talking about stability in this document as it's out of scope. If `stable` is not a good term here we can maybe find a better term which is a synonym of `-stable`?

Reply:
I cross checked other documents and we always refer to `stable`:
https://github.com/facebook/react-native/wiki/Deprecations

I couldn't find a better name and I think that `stable` here is clear enough.

I would also not add the sentence:
"Note that despite the -stable tag, there are no stability guarantees. Breaking changes will be announced in each release's notes."

as this goes in the direction of describing our API evolves, which should be moved to a separate document.